conans/test/functional/command/download_test.py

```diff
@@ -220,3 +220,57 @@ def no_user_channel_test(self):
         client.run("download pkg/1.0@")
         self.assertIn("pkg/1.0: Downloading pkg/1.0:%s" % NO_SETTINGS_PACKAGE_ID, client.out)
         self.assertIn("pkg/1.0: Package installed %s" % NO_SETTINGS_PACKAGE_ID, client.out)
+
+    def download_revs_disabled_with_rrev_test(self):
+        # https://github.com/conan-io/conan/issues/6106
+        client = TestClient(default_server_user=True, revisions_enabled=False)
```

Comment:
Tests that hardcode revisions should use something like:

```python
@unittest.skipIf(get_env("TESTING_REVISIONS_ENABLED", False), "This test is insane to be "
                                                                  "tested with revisions, in "
                                                                  "general all the module")
```
So they are only run in the configuration they want to run, not twice.